java/ql/src/Security/CWE/CWE-441/UnsafeContentUriResolution.qhelp

```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>
+  When an Android application wants to access data in a content provider, it uses the <code>ContentResolver</code>
+  object. <code>ContentResolver</code>s communicate with an instance of a class that implements the 
+  <code>ContentProvider</code> interface via URIs with the <code>content://</code> scheme.
+
+  The authority part (the first path segment) of the URI passed as parameter to the <code>ContentResolver</code> 
+  determines which content provider is contacted for the operation. Specific operations that act on files also
+  support the <code>file://</code> scheme, in which case the local filesystem is queried instead.
+
+  If an external component, like a malicious or compromised application, controls the URI that is used in a
+  <code>ContentResolver</code> operation, it can trick the vulnerable application into accessing its own private
+  files or non-exported content providers. Depending on what the vulnerable application does after accessing the file,
+  the attacking application might get access to the file by forcing it to be copied to a public directory like the
+  external storage, or tamper with it by making the application overwrite it with unexpected data.
```

Comment:
The last sentence here is difficult to parse, suggest simplifying it.

```suggestion
  If an external component, like a malicious or compromised application, controls the URI for a
  <code>ContentResolver</code> operation, it can trick the vulnerable application into accessing its own private
  files or non-exported content providers. The attacking application might be able to get access to the file by forcing it to be copied to a public directory, like
  external storage, or tamper with the contents by making the application overwrite the file with unexpected data.
```